src/jit/codegenxarch.cpp

```diff
@@ -185,16 +185,15 @@ void                CodeGen::genEmitGSCookieCheck(bool pushReg)
         // Handle multi-reg return type values
         if (compiler->compMethodReturnsMultiRegRetType())
         {
-#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
             ReturnTypeDesc retTypeDesc;
             retTypeDesc.Initialize(compiler, compiler->info.compMethodInfo->args.retTypeClass);
+            unsigned regCount = retTypeDesc.GetReturnRegCount();
+            assert(regCount == MAX_RET_REG_COUNT);
```

Comment:
Yes, for Arm, the corresponding assert will be in codegenarm.cpp/codegenarm64.cpp.  We need to add the right assert while adding multi-reg support on Arm32/arm64.  But since this is xarch, the above assert is valid.